lib/sharingmethodinfo.h

```diff
@@ -22,34 +23,18 @@
  * Lesser General Public License for more details.
  */
 
-#ifndef TRANSFERMETHODINFO_H
-#define TRANSFERMETHODINFO_H
+#ifndef SHARINGMETHODINFO_H
+#define SHARINGMETHODINFO_H
 
 #include <QtGlobal>
 #include <QtDBus/QtDBus>
 
-class TransferMethodInfoDeprecated
-{
-public:
-    friend QDBusArgument &operator<<(QDBusArgument &argument, const TransferMethodInfoDeprecated &record);
-    friend const QDBusArgument &operator>>(const QDBusArgument &argument, TransferMethodInfoDeprecated &record);
-
-    static void registerType();
-
-    QString displayName;
-    QString userName;
-    QString methodId;
-    QString shareUIPath;
-    QStringList capabilitities;
-    quint32 accountId;
-};
-
-class TransferMethodInfo
+class SharingMethodInfo
 {
 public:
 
     // Used internally only
-    enum TransferMethodInfoField {
+    enum SharingMethodInfoField {
         DisplayName,     // e.g. Facebook
         UserName,        // e.g. mike.myers@gmail.com
```

Comment:
While we are breaking the API, we could either add a separate sublabel property or then replace this with such. The account name is one case that benefits from extra label but sometimes apps can also provide different targets or mechanisms that can be used for sharing. Now they are abusing the username field :)

Similar way the AccountIcon is really used more like a generic sharing method icon. Could rename it.

For AccountId not sure is there reason to expose it from the plugins. Could be that the sharing method is using its internal concept of accounts.

Can be also adjusted separately though needs then a PR set for many of the same places as now.